nodegroups.tf

```diff
@@ -87,6 +87,22 @@ resource "aws_launch_template" "eks_node_groups" {
     enabled = true
   }
 
+  dynamic "instance_market_options" {
+    for_each = each.value.spot_nodes_enabled != null && each.value.spot_nodes_enabled == true ? ["do it"] : []
```

Comment:
@qedgardo why are we validating the same variable twice? shouldn't it be just 
`each.value.spot_nodes_enabled == true ? ["do it"] : []`

Reply:
This allows us to not be forced to set spot_nodes_enabled as false. If you don't set that attribute will work in the same way as the previous version. If you don't set true or false, the condition will fail because it will compare null with a boolean.